File: CS/Cars.UWP/EventArgsConverter.cs
using Cars.Model;
using DevExpress.Mvvm.UI;
using DevExpress.UI.Xaml.Scheduler;
using System.Linq;
using Windows.UI.Xaml.Markup;

namespace Cars.UWP {
    public class EventArgsConverter : MarkupExtension, IEventArgsConverter
    {

        protected override object ProvideValue()
        {
            return this;
        }

        public object Convert(object sender, object args)
        {
            if(args is AppointmentAddedEventArgs)
            {
                return (args as AppointmentAddedEventArgs).Appointments.FirstOrDefault().SourceObject;
            }

            if(args is AppointmentUpdatingEventArgs)
            {
                return (args as AppointmentUpdatingEventArgs).Appointments.FirstOrDefault().SourceObject;
            }

            if (args is AppointmentUpdatedEventArgs)
            {
                return (args as AppointmentUpdatedEventArgs).Appointments.FirstOrDefault().SourceObject;
            }

            if (args is AppointmentRemovingEventArgs)
            {
                return (args as AppointmentRemovingEventArgs).Appointments.Select(c => c.SourceObject as CarScheduling).ToList();
            }

            if(args is AppointmentRemovedEventArgs)
            {
                return (args as AppointmentRemovedEventArgs).Appointments.Select(c => c.SourceObject as CarScheduling).ToList();
            }

            return null;
        }
    }
}

File: VB/Cars.UWP/EventArgsConverter.vb
Imports Cars.Model
Imports DevExpress.Mvvm.UI
Imports DevExpress.UI.Xaml.Scheduler
Imports Windows.UI.Xaml.Markup

Public Class EventArgsConverter
    Inherits MarkupExtension
    Implements IEventArgsConverter

    Protected Overrides Function ProvideValue() As Object
        Return Me
    End Function

    Public Function Convert(ByVal sender As Object, ByVal args As Object) As Object Implements IEventArgsConverter.Convert
        If TypeOf args Is AppointmentAddedEventArgs Then
            Return (TryCast(args, AppointmentAddedEventArgs)).Appointments.FirstOrDefault().SourceObject
        End If

        If TypeOf args Is AppointmentUpdatingEventArgs Then
            Return (TryCast(args, AppointmentUpdatingEventArgs)).Appointments.FirstOrDefault().SourceObject
        End If

        If TypeOf args Is AppointmentUpdatedEventArgs Then
            Return (TryCast(args, AppointmentUpdatedEventArgs)).Appointments.FirstOrDefault().SourceObject
        End If

        If TypeOf args Is AppointmentRemovingEventArgs Then
            Return (TryCast(args, AppointmentRemovingEventArgs)).Appointments.[Select](Function(c) TryCast(c.SourceObject, CarScheduling)).ToList()
        End If

        If TypeOf args Is AppointmentRemovedEventArgs Then
            Return (TryCast(args, AppointmentRemovedEventArgs)).Appointments.[Select](Function(c) TryCast(c.SourceObject, CarScheduling)).ToList()
        End If

        Return Nothing
    End Function
End Class
